internal/util/util.go

```diff
@@ -60,15 +65,51 @@ func GetTempDir() string {
 }
 
 // ExportFromReader reads bytes from given reader and exports to external tar, directory or stdout.
-func ExportFromReader(input io.Reader, opts define.BuildOutputOption) error {
+func ExportFromReader(input io.ReadCloser, store storage.Store, opts define.BuildOutputOption) error {
 	var err error
 	if !filepath.IsAbs(opts.Path) {
 		opts.Path, err = filepath.Abs(opts.Path)
 		if err != nil {
 			return err
 		}
 	}
-	if opts.IsDir {
+	if opts.Type == "image" {
+		if opts.Attrs["push"] != "true" {
+			return nil
+		}
+
+		image := opts.Attrs["name"]
+		destSpec := opts.Attrs["name"]
+		dest, err := alltransports.ParseImageName(destSpec)
+		// add the docker:// transport to see if they neglected it.
+		if err != nil {
+			destTransport := strings.Split(destSpec, ":")[0]
+			if t := transports.Get(destTransport); t != nil {
+				return err
+			}
+
+			if strings.Contains(destSpec, "://") {
+				return err
+			}
+
+			destSpec = "docker://" + destSpec
+			dest2, err2 := alltransports.ParseImageName(destSpec)
+			if err2 != nil {
+				return err
+			}
+			dest = dest2
+			logrus.Debugf("Assuming docker:// as the transport method for DESTINATION: %s", destSpec)
+		}
+
+		libimageOptions := &libimage.PushOptions{}
+		libimageOptions.Writer = os.Stdout
+		runtime, err := libimage.RuntimeFromStore(store, &libimage.RuntimeOptions{SystemContext: &types.SystemContext{}})
+		destString := fmt.Sprintf("%s:%s", dest.Transport().Name(), dest.StringWithinTransport())
+		_, err = runtime.Push(context.Background(), image, destString, libimageOptions)
```

Comment:
(Eventually, could `libImage` have an API that accepts `type.ImageReference` directly?)